CMakeLists.txt

```
@@ -63,56 +67,46 @@ option(NAPI_BUILD_ABI "Build the ABI layer." ON)
 # Dependencies
 
 if(NOT TARGET arcana)
-    FetchContent_Declare(arcana
-        GIT_REPOSITORY https://github.com/microsoft/arcana.cpp.git
-        GIT_TAG 10d167ffe0f86b2ddaccd93fd78ff2240cbb0c49)
+    set(CONTENT_TO_FETCH ${CONTENT_TO_FETCH} arcana)
+endif()
 
-    message(STATUS "Fetching arcana.cpp")
-    FetchContent_MakeAvailable(arcana)
-    message(STATUS "Fetching arcana.cpp - done")
+if(NOT TARGET UrlLib AND JSRUNTIMEHOST_POLYFILL_XMLHTTPREQUEST)
+    set(CONTENT_TO_FETCH ${CONTENT_TO_FETCH} UrlLib)
+endif()
 
-    set_property(TARGET arcana PROPERTY FOLDER Dependencies)
+if(NAPI_JAVASCRIPT_ENGINE STREQUAL "V8" AND JSRUNTIMEHOST_CORE_APPRUNTIME_V8_INSPECTOR)
+    if(NOT TARGET asio)
+        set(CONTENT_TO_FETCH ${CONTENT_TO_FETCH} asio)
+    endif()
+
+    if(NOT TARGET llhttp_static)
+        set(BUILD_SHARED_LIBS OFF CACHE INTERNAL "")
+        set(BUILD_STATIC_LIBS ON CACHE INTERNAL "")
+        set(CONTENT_TO_FETCH ${CONTENT_TO_FETCH} llhttp)
+    endif()
 endif()
 
-if(NOT TARGET UrlLib AND JSRUNTIMEHOST_POLYFILL_XMLHTTPREQUEST)
-    FetchContent_Declare(
-        UrlLib
-        GIT_REPOSITORY https://github.com/BabylonJS/UrlLib.git
-        GIT_TAG 4182996cbe3b85a960e38bd8a61892688ed55213)
+# Fetching content
+message(STATUS "Fetching for ${CMAKE_PROJECT_NAME} : ${CONTENT_TO_FETCH}")
+FetchContent_MakeAvailable(${CONTENT_TO_FETCH})
+message(STATUS "${CMAKE_PROJECT_NAME} content fetched.")
```

See https://stackoverflow.com/questions/38938315/difference-between-cmake-project-name-and-project-name. I would also suggest we format it differently.

```suggestion
message(STATUS "Fetching dependencies for ${PROJECT_NAME} (${CONTENT_TO_FETCH})")
FetchContent_MakeAvailable(${CONTENT_TO_FETCH})
message(STATUS "Fetching dependencies for ${PROJECT_NAME} - done")
```